geomstats/geometry/discrete_surfaces.py

```diff
@@ -0,0 +1,687 @@
+            if self.d1 > 0 or self.b1 > 0 or self.a1 > 0:
+                ginv = gs.linalg.inv(g)
+                alpha1 = self.space.surface_one_forms(point1)
+                alpha2 = self.space.surface_one_forms(point2)
+                if self.d1 > 0:
+                    xi1 = alpha1 - alpha
+                    xi1_0 = gs.matmul(
+                        gs.matmul(alpha, ginv),
+                        gs.matmul(gs.transpose(xi1, (0, 2, 1)), alpha)
+                        - gs.matmul(gs.transpose(alpha, axes=(1, 2)), xi1),
+                    )
+                    xi2 = alpha2 - alpha
+                    xi2_0 = gs.matmul(
+                        gs.matmul(alpha, ginv),
+                        gs.matmul(gs.transpose(xi2, (0, 2, 1)), alpha)
+                        - gs.matmul(gs.transpose(alpha, axes=(1, 2)), xi2),
+                    )
+                    norm += self.d1 * gs.sum(
+                        gs.einsum(
+                            "bii->b",
+                            gs.matmul(
+                                xi1_0,
+                                gs.matmul(ginv, gs.transpose(xi2_0, axes=(0, 2, 1))),
+                            ),
+                        )
+                        * areas
+                    )
+                if self.b1 > 0 or self.a1 > 0:
+                    dg1 = gs.matmul(gs.transpose(alpha1, axes=(0, 2, 1)), alpha1) - g
+                    dg2 = gs.matmul(gs.transpose(alpha2, axes=(0, 2, 1)), alpha2) - g
+                    ginvdg1 = gs.matmul(ginv, dg1)
+                    ginvdg2 = gs.matmul(ginv, dg2)
+                    norm += self.a1 * gs.sum(
+                        gs.einsum("bii->b", gs.matmul(ginvdg1, ginvdg2)) * areas
+                    )
+                    norm += self.b1 * gs.sum(
+                        gs.einsum("bii->b", ginvdg1)
+                        * gs.einsum("bii->b", ginvdg2)
+                        * areas
+                    )
+        return norm
+
+    def squared_norm(self, vector, base_point):
+        """Squared norm of a tangent vector at a base point.
+
+        Parameters
+        ----------
+        vector: array-like, shape=[n_vertices, 3]
+            Tangent vector at base point.
+        base_point: array-like, shape=[n_vertices, dim]
+            Base point.
+
+        Returns
+        -------
+        squared_norm : float
+            Squared Norm.
+        """
+        return self.inner_product(vector, vector, base_point)
+
+    def stepwise_path_energy(self, path):
+        """Stepwise path energy of a PL path in the space of discrete surfaces.
+
+        Parameters
+        ----------
+        path: array-like, shape=[time_steps,n_vertices, 3]
+            PL path of discrete surfaces.
+
+        Returns
+        -------
+        stepwise_path_energy : array-like, shape=[time_steps-1]
+            Stepwise path energy.
+        """
+        N = path.shape[0]
```

Comment:
N --> n_times
Adapt this in the code and in the docstring

Reply:
I don't see "N" in the docstring, so i don't see where the docstring needs to be changed

Reply:
oh wait, i'm assuming you mean "time_steps" --> "n_times"